src/sdc11073/consumer/consumerimpl.py

```diff
@@ -552,24 +571,41 @@ def set_used_compression(self, *compression_methods: str):
         del self._compression_methods[:]
         self._compression_methods.extend(compression_methods)
 
-    def _get_metadata(self) -> mex_types.Metadata:
+    def _connect(self):
         _url = urlparse(self._device_location)
-        wsc = self.get_soap_client(self._device_location)
-
-        if self._ssl_context_container is not None and _url.scheme == 'https':
-            if wsc.is_closed():
-                wsc.connect()
-            sock = wsc.sock
+        if self.is_ssl_connection is not None:
+            # decision was already made in constructor
+            soap_client = self.get_soap_client(self._device_location)
```

Comment:
can be done before `if self.is_ssl_connection is not None:`

Reply:
done